src/ui/buttons/buttons.jsx

```diff
@@ -150,6 +152,47 @@ export function validateButtonProps(props: ButtonPropsInputs) {
   normalizeButtonProps(props);
 }
 
+function calculateMsgPosition({
+  message,
+  tagline,
+  layout,
+  fundingSources,
+  fundingSource,
+}): string {
+  if (!message) {
+    return null;
+  }
+  const { position } = message;
+  // Cases for throwing an error
+  if (tagline && layout === BUTTON_LAYOUT.HORIZONTAL && !fundingSource) {
+    throw new ValidationError("Button cannot show both Message and Tagline.");
+  }
+  if (
+    fundingSources.indexOf(FUNDING.CARD) !== -1 &&
+    position === MESSAGE_POSITION.BOTTOM
+  ) {
+    throw new ValidationError(
+      "Message position must be 'top' when Debit and/or Credit Card is a funding source"
+    );
+  }
+  // Position selection
+  if (
+    fundingSources.indexOf(FUNDING.CARD) !== -1 ||
+    position === MESSAGE_POSITION.TOP ||
+    (layout === BUTTON_LAYOUT.VERTICAL && !position)
+  ) {
+    return MESSAGE_POSITION.TOP;
+  } else if (
+    (layout === BUTTON_LAYOUT.VERTICAL &&
+      position === MESSAGE_POSITION.BOTTOM) ||
+    position === MESSAGE_POSITION.BOTTOM ||
+    layout === BUTTON_LAYOUT.HORIZONTAL ||
+    layout
+  ) {
+    return MESSAGE_POSITION.BOTTOM;
+  }
```

Comment:
Could this entire `else if` block be removed in favor of just having `return MESSAGE_POSTION.BOTTOM`? From the way this function appears to be structured, the top logic is responsible for determining if a message can be displayed, and this bottom section determines if it's top or bottom. If there's other scenarios where we'd prefer to return `null`, we should consider including them above. e.g. `if (...) { return null; }`

Reply:
this is true, I may have just included the bottom block so i could see that I factored in every condition while I was working, but now it's unnecessary. I'll remove that